packages/xchain-util/src/asset.ts

```diff
@@ -217,7 +226,17 @@ const NON_SYNTH_DELIMITER = '.'
  */
 export const assetFromString = (s: string): Asset | null => {
   const isSynth = s.includes(SYNTH_DELIMITER)
-  const delimiter = isSynth ? SYNTH_DELIMITER : NON_SYNTH_DELIMITER
+  let isTrade = s.includes(TRADE_DELIMITER)
+
+  isSynth ? true : (isTrade = false) // Can't be a Synth and Trade at the same time.
```

Comment:
The logic isSynth ? true : (isTrade = false) is problematic. It seems intended to ensure that an asset cannot be both a synth and a trade, but the implementation is flawed.

this is a refactored version 
export const assetFromString = (s: string): Asset | null => {
  const isSynth = s.includes(SYNTH_DELIMITER);
  const isTrade = !isSynth && s.includes(TRADE_DELIMITER); // Ensure isTrade is only considered if not a synth

  // Determine the delimiter based on whether the asset is a synth, trade, or neither
  const delimiter = isSynth ? SYNTH_DELIMITER : (isTrade ? TRADE_DELIMITER : NON_SYNTH_DELIMITER);

  const data = s.split(delimiter);
  if (data.length <= 1 || data[1]?.length < 1) {
    return null; 
  }
  return { chain, symbol, ticker, synth: isSynth, trade: isTrade };
};